src/mesh/distributed_mesh.C

```diff
@@ -167,6 +167,8 @@ DistributedMesh::DistributedMesh (const UnstructuredMesh & other_mesh) :
 #ifdef LIBMESH_ENABLE_UNIQUE_ID
   _next_unique_id = other_mesh.parallel_max_unique_id() +
                     this->processor_id();
+  _next_unpartitioned_unique_id = _next_unique_id +
+    (this->n_processors() - this->processor_id());
```

Comment:
Thanks. I think our code goes through` DistributedMesh::DistributedMesh (const DistributedMesh & other_mesh)` not this constructor. But I will try this out.

Reply:
Yeah, I think you're right.  But I didn't find anything else that might be getting used while uninitialized, so this was my best shot.